### ISSUE TITLE
azurerm_app_configuration_feature does not work properly if filters are not provided 

### ISSUE THREAD

**Issue by my3sons** [hashicorp/terraform-provider-azurerm#16329]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.5

### AzureRM Provider Version

2.98.0

### Affected Resource(s)/Data Source(s)

azurerm_app_configuration_feature

### Terraform Configuration Files

```hcl
terraform {
  experiments = [module_variable_optional_attrs]
}

locals {
  configuration_store_id = var.configuration_store_id
  features = [for feature in var.app_config_features :
    defaults(feature, {
      enabled     = false
      description = ""
    })
  ]
}

resource "azurerm_app_configuration_feature" "test" {
  configuration_store_id = local.configuration_store_id
  for_each               = { for feature in local.features : feature.name => feature }
  description            = each.value.description
  name                   = each.key
  label                  = each.value.label
  enabled                = each.value.enabled
}
```

### Debug Output/Panic Output

```shell
Exception while parsing feature flag. key:.appconfig.featureflag/myFeatureFlag value:{"conditions":{"client_filters":null},"description":"","enabled":true,"id":"myFeatureFlag"}.
'NoneType' object is not iterable
```

### Expected Behaviour

The expected/required behavior is that the above config would create a feature in App Configuration that looks like the following when edited via the Azure Portal:

```
{
  "conditions": {
    "client_filters": []
  },
  "description": "",
  "enabled": true,
  "id": "myFeatureFlag"
}
```

### Actual Behaviour

However, what I actually see is the following.. NOTE the **null** value for client_filters. 

```
{
  "conditions": {
    "client_filters": null
  },
  "description": "",
  "enabled": true,
  "id": "myFeatureFlag"
}
```

As a result, you are not able to query the App Configuration resource in any way for features, nor do I think you can access the features via Azure application SDKs. For example, if I run the following query:

`az appconfig feature list --name myAppConfigurationResource`

The following error is returned:
```
Exception while parsing feature flag. key:.appconfig.featureflag/myFeatureFlag value:{"conditions":{"client_filters":null},"description":"","enabled":true,"id":"myFeatureFlag"}.
'NoneType' object is not iterable
```

If I go in and replace the null value for client_filters to [], everything starts working again.

I have been able to work around this for now by using the azurerm_app_configuration_key resource and doing the following:

```
resource "azurerm_app_configuration_key" "this" {
  configuration_store_id = local.configuration_store_id
  for_each               = { for feature in local.features : feature.name => feature }
  key                    = ".appconfig.featureflag/${each.key}"
  label                  = each.value.label
  value                  = jsonencode({ "id" = "${each.key}", "description" = "${each.value.description}", "enabled" = "${each.value.enabled}", "conditions" = { client_filters = [] } })
  content_type           = "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
}
```

While this works, I would prefer to use the intended resource for features.

**Comment by MikeRhoden** [comment]
I came here to report same issue.  Thanks for @my3sons for the great description and work around.

**Comment by sinbai** [comment]
@my3sons thank you for opening this issue. I have submitted PR #16459 to fix this issue. Please track that to see whether you would be unblocked when it is released.

**Comment by stephybun** [comment]
Closed by #16459
